pkg/controller/userreport/send.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userreport
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/google/exposure-notifications-server/pkg/base64util"
+	"github.com/google/exposure-notifications-server/pkg/logging"
+	"github.com/google/exposure-notifications-verification-server/pkg/api"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller/issueapi"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/gorilla/sessions"
+)
+
+func stringFromSession(session *sessions.Session, key string) (string, error) {
+	v, ok := session.Values[key]
+	if !ok {
+		return "", fmt.Errorf("there is no %q in the session", key)
+	}
+	strVal, ok := v.(string)
+	if !ok {
+		return "", fmt.Errorf("the session value for %q is not a string", key)
+	}
+	return strVal, nil
+}
+
+func (c *Controller) HandleSend() http.Handler {
+	type FormData struct {
+		TestDate    string `form:"testDate"`
+		SymptomDate string `form:"symptomDate"`
+		Phone       string `form:"phone"`
+		Agreement   bool   `form:"agreement"`
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		logger := logging.FromContext(ctx).Named("userreport.HandleSend")
+
+		session := controller.SessionFromContext(ctx)
+		if session == nil {
+			logger.Debugw("missing session")
```

Comment:
You don't need these debug statements - the middleware will log the correct thing